Mathlib/Tactic/Check.lean

```diff
@@ -17,24 +17,56 @@ since it elaborates `t` in a more tolerant way and so it can be possible to get
 For example, `#check` allows metavariables.
 -/
 
+open Lean Elab Meta Tactic
+
+open Command PrettyPrinter Delaborator in
+/-- The `#check'` command is like `#check`, but only prints explicit arguments in the signature
+(i.e., omitting implicit and typeclass arguments). -/
+elab tk:"#check' " name:ident : command => runTermElabM fun _ => do
+  for c in (← realizeGlobalConstWithInfos name) do
+    addCompletionInfo <| .id name name.getId (danglingDot := false) {} none
+    let info ← getConstInfo c
+    let delab : Delab := do
+      delabForallParamsWithSignature fun binders type => do
+        let binders := binders.filter fun binder => binder.raw.isOfKind ``Parser.Term.explicitBinder
+        return ⟨← `(declSigWithId| $(mkIdent c) $binders* : $type)⟩
+    logInfoAt tk <| .ofFormatWithInfosM (PrettyPrinter.ppExprWithInfos (delab := delab) info.type)
+
 namespace Mathlib.Tactic
 
-open Lean Meta Elab Tactic
+/-- Core routine for the `#check` tactic: show a signature for `#check term`, assuming `term`
+is an identifier. Info messages are placed at `tk`.
+In case there are several resolved names for `term`, show information for all of them. -/
+def checkInner (tk : Syntax) (term : Term) : TacticM Unit := do
+  for c in (← realizeGlobalConstWithInfos term) do
+    addCompletionInfo <| .id term c (danglingDot := false) {} none
+    logInfoAt tk <| MessageData.signature c
+    return
 
-/--
-Tactic version of `Lean.Elab.Command.elabCheck`.
-Elaborates `term` without modifying tactic/elab/meta state.
-Info messages are placed at `tk`.
--/
-def elabCheckTactic (tk : Syntax) (ignoreStuckTC : Bool) (term : Term) : TacticM Unit :=
+open PrettyPrinter Delaborator in
+/-- Core routine for the `#check'` tactic: show a signature for `#check' term`, assuming `term`
+is an identifier. Info messages are placed at `tk`.
+In case there are several resolved names for `term`, show information for all of them. -/
+def checkPrimeInner (tk : Syntax) (term : Term) : TacticM Unit := do
+  for c in (← realizeGlobalConstWithInfos term) do
+    addCompletionInfo <| .id term c (danglingDot := false) {} none
+    let info ← getConstInfo c
+    let delab : Delab := do
+      delabForallParamsWithSignature fun binders type => do
+        let binders := binders.filter fun binder => binder.raw.isOfKind ``Parser.Term.explicitBinder
+        return ⟨← `(declSigWithId| $(mkIdent c) $binders* : $type)⟩
+    logInfoAt tk <| .ofFormatWithInfosM (PrettyPrinter.ppExprWithInfos (delab := delab) info.type)
+    return
+
+/-- Workhorse method for the `#check` and `#check'` tactic.
+This does all the set-up; the actual behaviour is governed by the function `inner` passed in. -/
+def elabCheckTacticInner (tk : Syntax) (ignoreStuckTC : Bool) (term : Term)
```

Comment:
Minor, but I think let's allow the `inner` functions to have the `inner` names to avoid confusion! :) (Alternate suggestion: `elabCheckTacticOfInner`)
```suggestion
def elabCheckTacticAux (tk : Syntax) (ignoreStuckTC : Bool) (term : Term)
```